configure.ac

```diff
@@ -102,6 +102,14 @@ AS_IF([test "x$enable_test" = xyes || test "x$enable_compare" = xyes], [
     ])
 ])
 
+pthread=false
+AX_PTHREAD([pthread=true])
+AC_CHECK_HEADER([stdatomic.h], [], [pthread=false])
+if $pthread; then
+	AC_DEFINE(CONFIG_PTHREAD, 1, [Have POSIX threads])
+fi
+AM_CONDITIONAL(CONFIG_PTHREAD, $pthread)
+
```

Comment:
From a first look and test, this fails to build when any utilties like are enabled as it doesn't properly add the needed flags. Not sure why it even works for the shared lib build *(does linking with the resulting shared lib work for you?)*. Also e.g. JSO will want to disable pthreads even if it is available at build time.
With the below change it should be better, though some questions remain.
```patch
From 105cad27c3f00a7ab9602953a91a7b7dbbd8506d Mon Sep 17 00:00:00 2001
From: Oneric <oneric@oneric.stub>
Date: Fri, 29 Jul 2022 19:06:50 +0200
Subject: [PATCH] fix threading config

 - The libs and flags were not added to the build making utility builds
   fail (not sure why it even worked for shared lib)
 - Flags also need to be added to pc-file
 - Automake conditional was defined but never used
 - Move header check to HEADERS_ONCE for better order and perf
 - when building for special envs like in-browser WASM, one may wish to
   disable threading even if (sometimes) techincally possible

Not sure if putting only PTHREAD_LIBS (here: -lpthread) into the
pc-file's Lib.private is good enough or if PTHREAD_CFLAGS (here:
-pthread) should also be involved.

Checking pc-files on my system only 2 (gthread-2.0,
gmodule-no-export) add -pthread to their Cflags and the same two
additionally added -pthread to their (public) Libs.
4 more add -lpthread to the public libs.
46 add either -pthread, -lpthread or both to Libs.private.
---
 configure.ac | 24 ++++++++++++++++--------
 1 file changed, 16 insertions(+), 8 deletions(-)

diff --git a/configure.ac b/configure.ac
index b165f5e..96d5d6c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -10,7 +10,7 @@ AC_PROG_CC
 AM_PROG_CC_C_O
 
 # Checks for header files.
-AC_CHECK_HEADERS_ONCE([iconv.h])
+AC_CHECK_HEADERS_ONCE([iconv.h stdatomic.h])
 
 # Checks for library functions.
 AC_CHECK_FUNCS([strdup strndup])
@@ -34,6 +34,8 @@ AC_ARG_ENABLE([require-system-font-provider], AS_HELP_STRING([--disable-require-
     [allow compilation even if no system font provider was found @<:@default=enabled:>@]))
 AC_ARG_ENABLE([asm], AS_HELP_STRING([--disable-asm],
     [disable compiling with ASM @<:@default=check@:>@]))
+AC_ARG_ENABLE([threads], AS_HELP_STRING([--disable-threads],
+    [disable threading support @<:@default=check@:>@]))
 AC_ARG_ENABLE([large-tiles], AS_HELP_STRING([--enable-large-tiles],
     [use larger tiles in the rasterizer (better performance, slightly worse quality) @<:@default=disabled@:>@]))
 
@@ -102,13 +104,19 @@ AS_IF([test "x$enable_test" = xyes || test "x$enable_compare" = xyes], [
     ])
 ])
 
-pthread=false
-AX_PTHREAD([pthread=true])
-AC_CHECK_HEADER([stdatomic.h], [], [pthread=false])
-if $pthread; then
-   AC_DEFINE(CONFIG_PTHREAD, 1, [Have POSIX threads])
-fi
-AM_CONDITIONAL(CONFIG_PTHREAD, $pthread)
+AS_IF([test "x$enable_threads" != xno && test "x$ac_cv_header_stdatomic_h" = xyes], [
+    AX_PTHREAD([
+        pkg_libs="$pkg_libs $PTHREAD_LIBS"
+        CFLAGS="$CFLAGS $PTHREAD_CFLAGS"
+        LIBS="$LIBS $PTHREAD_LIBS"
+        CC="$PTHREAD_CC"
+        AC_DEFINE(CONFIG_PTHREAD, 1, [Have POSIX threads])
+    ], [
+        AS_IF([test "x$enable_threads" = xyes], [
+            AC_MSG_ERROR([threading support was requested, but it was not found.])
+        ])
+    ])
+])
 
 ## Check for system font providers
 ### Fontconfig
-- 
2.30.2
```

Reply:
Actually, this will fail to error out if threading was explicitly requested but `stdatomic.h` not found, so the conditionals need to be moved a bit and maybe a temporary `can_threads` var will be needed.

Reply:
On another thought, automatically replacing `CC` when a special compiler is required for pthreads may be surprising if the user explicitly defined `CC` previously. This (currently) only affects AIX systems with their default (non-GCC) compiler though. Maybe just printing an error instead of automatically reassinging `CC` is good enough?
```autoconf
AS_IF([test "$PTHREAD_CC" != "$CC"], [
    AC_MSG_ERROR(m4_join(m4_newline([                  ]),
        [Chosen or default compiler is unable to use pthreads, but another one is.],
        [Chosen/Default: "$CC"],
        [Expected for threads: "$PTHREAD_CC"]
    ))
 ])
```

Reply:
I think this should be fine; all the macro does is append `_r` to the already-selected `CC` (and only if `${CC}_r` is a usable executable and we're not using something GCC-compatible… which makes me wonder if this case would even build libass anyway). It'd be pretty weird to have a `_r` alias there and not be expecting it to be used by this kind of thing.

Anyway, I tweaked your changeset a little and merged it into the pthread-check commit, and all seems well in the linux builds now; thanks much.